builtin/credential/cert/path_login.go

```diff
@@ -155,7 +155,7 @@ func validateConnState(roots *x509.CertPool, cs *tls.ConnectionState) ([][]*x509
 		KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
 	}
 	certs := cs.PeerCertificates
-	if len(certs) == 0 {
+	if certs == nil || len(certs) == 0 {
```

Comment:
Hm. Upon further testing you appear to be correct.

However, that was the line that the traceback was pointing to, and that was the only thing that it could possibly have been on that line. It's possible the guy was running an old build and the line numbers are no longer accurate.